slsDetectorSoftware/src/CmdParser.cpp

```diff
@@ -36,64 +37,75 @@ void CmdParser::Parse(int argc, const char *const argv[]) {
     }
 }
 
-void CmdParser::Parse(const std::string &s) {
+void CmdParser::Parse(std::string s) {
+    //taking s by value we can modify it. 
     Reset();
+
+    //Are we looking at -h --help?
+    bool h = replace_first(&s, "--help", " ");
+    h = h || replace_first(&s, "-h", " ");
+    help_ = h;
+
+    // Extract the position indicies
+    auto pos = s.find_first_not_of("0123456789:- ");
+    if (pos!=0){
+        auto pre = s.substr(0, pos);
+        pre.erase(std::remove(pre.begin(), pre.end(), ' '), pre.end());
+        s.erase(0, pos);
+        DecodeIdAndPosition(pre.c_str());
+    }
+    
+    //Command and args should now be all that's left in the string
     std::istringstream iss(s);
     auto it = std::istream_iterator<std::string>(iss);
+    command_ = *it++; //First arg is the comand to run
+
     arguments_ =
         std::vector<std::string>(it, std::istream_iterator<std::string>());
-    auto old_size = arguments_.size();
-    arguments_.erase(std::remove_if(begin(arguments_), end(arguments_),
-                                    [](const std::string &item) {
-                                        return (item == "-h" ||
-                                                item == "--help");
-                                    }),
-                     end(arguments_));
-    if (old_size - arguments_.size() > 0)
-        help_ = true;
-    if (!arguments_.empty()) {
-        command_ = arguments_[0];
-        arguments_.erase(begin(arguments_));
-    }
+
     //allow comma sep
```

Comment:
Added to support p counters 1, 2 etc. Since people (at least I) tend to add the commas out of habit. But could be removed, in the current implementation it still required the space.